### DISCUSSION TITLE
[Question]: 阿里云百炼平台发布的qwen2.5-72B和开源qwen2.5-72B能力差距大

### DISCUSSION THREAD

**Discussion by darvsum** [QwenLM/Qwen3#964] [General]
### Has this been raised before?

- [X] I have checked [the GitHub README](https://github.com/QwenLM/Qwen2.5).
- [X] I have checked [the Qwen documentation](https://qwen.readthedocs.io) and cannot find an answer there.
- [X] I have searched [the issues](https://github.com/QwenLM/Qwen2.5/issues?q=is%3Aissue) and there is not a similar one.
- [X] I confirm that this is not a bug report, a feature request, or a badcase.

### Description

对比了一下阿里云百炼平台和开源的qwen2.5-72B。使用两个模型分别抽取知识图谱，开源的只能抽取出节点，关系一条都没抽取出来，而百炼平台的模型可以抽取出节点和关系。是不是说明开源的qwen2.5-72B做了简化处理？

**Comment by jklj077** [comment]
Did you use the instruct models? The models are the same. The difference should mainly came from how you run the model.

**Comment by monkeywl2020** [comment]
> 台的模型可以抽取出节点和关系。是不是说明开源的qwen2.5-72B做了简化处理？

我试过，可以抽取知识图谱的关系节点，可能需要修改下提示词，并且修改下返回内容的解析函数就可以了。这部分需要自己适配。

**Comment by darvsum** [comment]
#>Did you use the instruct models? The models are the same. The difference should mainly came from how you run the model.
两个都是使用的 instruct models，代码也是一样的。 @jklj077

**Comment by darvsum** [comment]
@monkeywl2020 ，我用的graph_transformers默认的提示词。开源模型换成qwen2-7b-instruct竟然抽取实体比qwen2.5-72b-instruct效果好。但是几个开源模型抽取关系始终是空值。

**Comment by jklj077** [comment]
langchain? what did you actually use?

**Reply by darvsum** [reply]
@jklj077 
是使用的langchain。问题找到了，归根还是开源模型能力弱了。使用的是这个例子https://python.langchain.com/docs/how_to/graph_constructing/

函数调用返回值：
{'raw': AIMessage(content='', additional_kwargs={'tool_calls': [{'id': 'call_5cr7krxq', 'function': {'arguments': '{"nodes":[{"id":"Marie Curie","type":"person"},{"id":"Pierre Curie","type":"person"},{"id":"Nobel Prize","type":"award"},{"id":"University of Paris","type":"university"},{"id":"Polish","type":"nationality"},{"id":"French","type":"nationality"},{"id":"physics","type":"field_of_study"},{"id":"chemistry","type":"field_of_study"},{"id":"radioactivity","type":"concept"}],"relationships":[{"source":"Marie Curie","target":"Nobel Prize","type":"RECIPIENT_OF_AWARD"},{"source":"Marie Curie","target":"University of Paris","type":"PROFESSOR_AT"},{"source":"Marie Curie","target":"Polish","type":"HAS_NATIONALITY"},{"source":"Marie Curie","target":"French","type":"HAS_NATIONALITY"},{"source":"Marie Curie","target":"physics","type":"STUDIED_FIELD"},{"source":"Marie Curie","target":"chemistry","type":"STUDIED_FIELD"},{"source":"Marie Curie","target":"radioactivity","type":"RESEARCHED_ON"},{"source":"Pierre Curie","target":"Nobel Prize","type":"RECIPIENT_OF_AWARD"},{"source":"Marie Curie","target":"Pierre Curie","type":"MARRIED_TO"}]}', 'name': 'DynamicGraph'}, 'type': 'function'}], 'refusal': None}, response_metadata={'token_usage': {'completion_tokens': 389, 'prompt_tokens': 709, 'total_tokens': 1098}, 'model_name': 'qwen2:72b-instruct-fp16', 'system_fingerprint': 'fp_ollama', 'finish_reason': 'tool_calls', 'logprobs': None}, id='run-4979304b-703d-4549-b500-cc16355cc932-0', tool_calls=[{'name': 'DynamicGraph', 'args': {'nodes': [{'id': 'Marie Curie', 'type': 'person'}, {'id': 'Pierre Curie', 'type': 'person'}, {'id': 'Nobel Prize', 'type': 'award'}, {'id': 'University of Paris', 'type': 'university'}, {'id': 'Polish', 'type': 'nationality'}, {'id': 'French', 'type': 'nationality'}, {'id': 'physics', 'type': 'field_of_study'}, {'id': 'chemistry', 'type': 'field_of_study'}, {'id': 'radioactivity', 'type': 'concept'}], 'relationships': [{'source': 'Marie Curie', 'target': 'Nobel Prize', 'type': 'RECIPIENT_OF_AWARD'}, {'source': 'Marie Curie', 'target': 'University of Paris', 'type': 'PROFESSOR_AT'}, {'source': 'Marie Curie', 'target': 'Polish', 'type': 'HAS_NATIONALITY'}, {'source': 'Marie Curie', 'target': 'French', 'type': 'HAS_NATIONALITY'}, {'source': 'Marie Curie', 'target': 'physics', 'type': 'STUDIED_FIELD'}, {'source': 'Marie Curie', 'target': 'chemistry', 'type': 'STUDIED_FIELD'}, {'source': 'Marie Curie', 'target': 'radioactivity', 'type': 'RESEARCHED_ON'}, {'source': 'Pierre Curie', 'target': 'Nobel Prize', 'type': 'RECIPIENT_OF_AWARD'}, {'source': 'Marie Curie', 'target': 'Pierre Curie', 'type': 'MARRIED_TO'}]}, 'id': 'call_5cr7krxq', 'type': 'tool_call'}], usage_metadata={'input_tokens': 709, 'output_tokens': 389, 'total_tokens': 1098}), 'parsing_error': 36 validation errors for DynamicGraph
relationships.0.source_node_id
  Field required [type=missing, input_value={'source': 'Marie Curie',...': 'RECIPIENT_OF_AWARD'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.0.source_node_type
  Field required [type=missing, input_value={'source': 'Marie Curie',...': 'RECIPIENT_OF_AWARD'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.0.target_node_id
  Field required [type=missing, input_value={'source': 'Marie Curie',...': 'RECIPIENT_OF_AWARD'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.0.target_node_type
  Field required [type=missing, input_value={'source': 'Marie Curie',...': 'RECIPIENT_OF_AWARD'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.1.source_node_id
  Field required [type=missing, input_value={'source': 'Marie Curie',... 'type': 'PROFESSOR_AT'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.1.source_node_type
  Field required [type=missing, input_value={'source': 'Marie Curie',... 'type': 'PROFESSOR_AT'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.1.target_node_id
  Field required [type=missing, input_value={'source': 'Marie Curie',... 'type': 'PROFESSOR_AT'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.1.target_node_type
  Field required [type=missing, input_value={'source': 'Marie Curie',... 'type': 'PROFESSOR_AT'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.2.source_node_id
  Field required [type=missing, input_value={'source': 'Marie Curie',...ype': 'HAS_NATIONALITY'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.2.source_node_type
  Field required [type=missing, input_value={'source': 'Marie Curie',...ype': 'HAS_NATIONALITY'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.2.target_node_id
  Field required [type=missing, input_value={'source': 'Marie Curie',...ype': 'HAS_NATIONALITY'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.2.target_node_type
  Field required [type=missing, input_value={'source': 'Marie Curie',...ype': 'HAS_NATIONALITY'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.3.source_node_id
  Field required [type=missing, input_value={'source': 'Marie Curie',...ype': 'HAS_NATIONALITY'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.3.source_node_type
  Field required [type=missing, input_value={'source': 'Marie Curie',...ype': 'HAS_NATIONALITY'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.3.target_node_id
  Field required [type=missing, input_value={'source': 'Marie Curie',...ype': 'HAS_NATIONALITY'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.3.target_node_type
  Field required [type=missing, input_value={'source': 'Marie Curie',...ype': 'HAS_NATIONALITY'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.4.source_node_id
  Field required [type=missing, input_value={'source': 'Marie Curie',...'type': 'STUDIED_FIELD'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.4.source_node_type
  Field required [type=missing, input_value={'source': 'Marie Curie',...'type': 'STUDIED_FIELD'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.4.target_node_id
  Field required [type=missing, input_value={'source': 'Marie Curie',...'type': 'STUDIED_FIELD'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.4.target_node_type
  Field required [type=missing, input_value={'source': 'Marie Curie',...'type': 'STUDIED_FIELD'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.5.source_node_id
  Field required [type=missing, input_value={'source': 'Marie Curie',...'type': 'STUDIED_FIELD'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.5.source_node_type
  Field required [type=missing, input_value={'source': 'Marie Curie',...'type': 'STUDIED_FIELD'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.5.target_node_id
  Field required [type=missing, input_value={'source': 'Marie Curie',...'type': 'STUDIED_FIELD'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.5.target_node_type
  Field required [type=missing, input_value={'source': 'Marie Curie',...'type': 'STUDIED_FIELD'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.6.source_node_id
  Field required [type=missing, input_value={'source': 'Marie Curie',...'type': 'RESEARCHED_ON'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.6.source_node_type
  Field required [type=missing, input_value={'source': 'Marie Curie',...'type': 'RESEARCHED_ON'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.6.target_node_id
  Field required [type=missing, input_value={'source': 'Marie Curie',...'type': 'RESEARCHED_ON'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.6.target_node_type
  Field required [type=missing, input_value={'source': 'Marie Curie',...'type': 'RESEARCHED_ON'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.7.source_node_id
  Field required [type=missing, input_value={'source': 'Pierre Curie'...': 'RECIPIENT_OF_AWARD'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.7.source_node_type
  Field required [type=missing, input_value={'source': 'Pierre Curie'...': 'RECIPIENT_OF_AWARD'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.7.target_node_id
  Field required [type=missing, input_value={'source': 'Pierre Curie'...': 'RECIPIENT_OF_AWARD'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.7.target_node_type
  Field required [type=missing, input_value={'source': 'Pierre Curie'...': 'RECIPIENT_OF_AWARD'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.8.source_node_id
  Field required [type=missing, input_value={'source': 'Marie Curie',...', 'type': 'MARRIED_TO'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.8.source_node_type
  Field required [type=missing, input_value={'source': 'Marie Curie',...', 'type': 'MARRIED_TO'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.8.target_node_id
  Field required [type=missing, input_value={'source': 'Marie Curie',...', 'type': 'MARRIED_TO'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing
relationships.8.target_node_type
  Field required [type=missing, input_value={'source': 'Marie Curie',...', 'type': 'MARRIED_TO'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/missing, 'parsed': None}

模型调用函数虽然有返回值，但是relationships的返回值格式不正确，导致为空。百炼平台线上的相同参数量的模型调用没有问题。能否优化一下？

**Reply by jklj077** [reply]
it's the same model. wait did you use ollama?

**Reply by jklj077** [reply]
![image](https://github.com/user-attachments/assets/a704fbf5-2991-41af-b9dd-0825eaaacc74)

This is result from using the OpenAI-compatible API from vLLM. It just works.

**Reply by darvsum** [reply]
是的，使用了ollama，但还是用其他兼容openai的本地api服务，结果都一样。没有试vllm，我试试。

**Reply by darvsum** [reply]
@jklj077 经过验证使用vllm可行。感谢！
1、使用的是相同的模型，运行多次，节点抽取数始终就这几个，没有你的多
![Snipaste_2024-09-26_16-21-02](https://github.com/user-attachments/assets/1e1d5e2f-da1d-4fa6-b3a1-d3a250610bb8)

2、node_properties对应的值本地模型没有，线上有

**Reply by darvsum** [reply]
我更换模型为qwen2-72b-instruct，达到了满意结果。是不是说明qwen2.5在文本理解方面能力减弱了呢？
